### PR TITLE
Set taxon left right values

### PR DESCRIPTION
**Pull request by ctSkennerton**
This pull request adds the ability to set/update the left and right values of the taxonomy table when a new taxonomy is added in. This is useful when the database has not been created using the `load_ncbi_taxonomy.pl` script or when a new species (or other taxon) is added into a database with an outdated copy of the NCBI taxonomy loaded into it. As a side effect to this I also added in an `executemany` method to the database adaptor as it makes the code for updating the left and right values much easier. I think that executemany is required in the python DBI 2.0 spec so there shouldn't be a problem with the underlying DBI adaptors not implementing it.

### CODE REVIEW

File: BioSQL/Loader.py [hunk 1]
```diff
@@ -329,6 +329,46 @@ def add_space(letter):
         assert answer == answer.lower()
         return answer
 
+    def _update_left_right_taxon_values(self, left_value):
+        """ update the left and right values in the table
```
Comment (peterjc) [inline BioSQL/Loader.py]: Remove leading space on docstring?

File: BioSQL/DBUtils.py [hunk 2]
```diff
@@ -40,6 +40,11 @@ def execute(self, cursor, sql, args=None):
         """
         cursor.execute(sql, args or ())
 
+    def executemany(self, cursor, sql, seq):
+        """
```
Comment (peterjc) [inline BioSQL/DBUtils.py]: No docstring?

**Comment by peterjc** [comment]
This looks sensible, not sure who else could review it - @hlapp any comments?

**Comment by ctSkennerton** [comment]
updated the docstrings 😄

**Comment by hlapp** [comment]
I'm not sure I can comment competently. The enumeration implementation seems to work a little differently from the default, so for example I don't know where the increment by +2 is coming from, and other little details. In theory well written tests should show that it's correct though 😃

**Comment by ctSkennerton** [comment]
I believe that all of the tests now pass, however it took a number of additional changes:
1. I had to remove an assertion that had compatibility problems between python 3 and python 2 as it was checking for the long type.
2. Provide an `executemany` method to the `_CursorWrapper` class, which is specific to the mysql-connector 
3. Change one of the SQL statements in the test file that was causing problems for Postgres

**Comment by peterjc** [comment]
One TravisCI build failed due to a dependency issue, rerunning it now.

QuantifiedCode looks OK, other than on ``execturemany`` method still missing a docstring.

File: BioSQL/BioSeqDatabase.py [hunk 3]
```diff
@@ -313,6 +313,9 @@ def __init__(self, real_cursor):
     def execute(self, operation, params=None, multi=False):
         self.real_cursor.execute(operation, params, multi)
 
+    def executemany(self, operation, params):
+        self.real_cursor.executemany(operation, params)
```
Comment (peterjc) [inline BioSQL/BioSeqDatabase.py]: Missing docstring e.g.

```Execute many SQL commands.```

File: BioSQL/BioSeqDatabase.py [hunk 4]
```diff
@@ -489,6 +492,13 @@ def execute(self, sql, args=None):
             sql = sql.replace("%s", "?")
         self.dbutils.execute(self.cursor, sql, args)
 
+    def executemany(self, sql, args):
+        """Just execute an sql command.
```
Comment (peterjc) [inline BioSQL/BioSeqDatabase.py]: Many commands? e.g.

```Execute many SQL commands.```

**Comment by ctSkennerton** [comment]
fixed up the docstrings as requested!

**Comment by peterjc** [comment]
Thanks @ctSkennerton, and @hlapp for your input. I've applied this as a squashed commit which also fixed the PEP8 whitespace complaints.

**Pull request closed without merging**